modules/imgproc/src/corner.cpp

```diff
@@ -67,7 +140,22 @@ static void calcMinEigenVal( const Mat& _cov, Mat& _dst )
         const float* cov = _cov.ptr<float>(i);
         float* dst = _dst.ptr<float>(i);
         j = 0;
-    #if CV_SIMD128
+#if CV_AVX
+        if( simd )
+        {
+            __m256 half = _mm256_set1_ps(0.5f);
+            for( ; j <= size.width - 8; j += 8 )
+            {
+                __m256 v_a, v_b, v_c, v_t;
+                load_deinterleave(cov + j*3, v_a, v_b, v_c);
+                v_a = _mm256_mul_ps(v_a, half);
+                v_c = _mm256_mul_ps(v_c, half);
+                v_t = _mm256_sub_ps(v_a, v_c);
+                v_t = _mm256_add_ps(_mm256_mul_ps(v_b, v_b), _mm256_mul_ps(v_t, v_t));
+                _mm256_storeu_ps(dst + j, _mm256_sub_ps(_mm256_add_ps(v_a, v_c), _mm256_sqrt_ps(v_t)));
+            }
+        }
+#elif CV_SIMD128
```

Comment:
Why is "#elif" here?
If AVX is not available during runtime (but we compile with AVX), then we probably want to try SSE instructions too ("simd" variable should be spitted too).